sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/ZPageServer.java

```diff
@@ -0,0 +1,165 @@
+ * <pre>{@code
+ * public class Main {
+ *   public static void main(String[] args) throws Exception {
+ *     ZPageServer.startHttpServerAndRegisterAllPages(8000);
+ *     ... // do work
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>Example usage with shared {@link HttpServer}
+ *
+ * <pre>{@code
+ * public class Main {
+ *   public static void main(String[] args) throws Exception {
+ *     HttpServer server = HttpServer.create(new InetSocketAddress(8000), 10);
+ *     ZPageServer.registerAllPagesToHttpServer(server);
+ *     server.start();
+ *     ... // do work
+ *   }
+ * }
+ * }</pre>
+ */
+@ThreadSafe
+public final class ZPageServer {
+  // The maximum number of queued incoming connections allowed on the HttpServer listening socket.
+  private static final int HTTPSERVER_BACKLOG = 5;
+  // Length of time to wait for the HttpServer to stop
+  private static final int HTTPSERVER_STOP_DELAY = 1;
+  // Tracez SpanProcessor and DataAggregator for constructing TracezZPageHandler
+  private static final TracezSpanProcessor tracezSpanProcessor =
+      TracezSpanProcessor.newBuilder().build();
+  private static final TracezDataAggregator tracezDataAggregator =
+      new TracezDataAggregator(tracezSpanProcessor);
+  // Handler for /tracez page
+  private static final ZPageHandler tracezZPageHandler =
+      new TracezZPageHandler(tracezDataAggregator);
+
+  private static final Object mutex = new Object();
+  private static volatile AtomicBoolean isTracezSpanProcesserAdded = new AtomicBoolean(false);
+
+  @GuardedBy("mutex")
+  @Nullable
+  private static HttpServer server;
+
+  /** Function that adds the {@link TracezSpanProcessor} to the {@link tracerSdkProvider}. */
+  private static void addTracezSpanProcessor() {
+    if (isTracezSpanProcesserAdded.compareAndSet(false, true)) {
+      synchronized (mutex) {
+        TracerSdkProvider tracerProvider = OpenTelemetrySdk.getTracerProvider();
+        tracerProvider.addSpanProcessor(tracezSpanProcessor);
+      }
+    }
+  }
+
+  /**
+   * Returns a {@code ZPageHandler} for tracing debug. The page displays information about all
+   * running spans and all sampled spans based on latency and error.
+   *
+   * <p>It displays a summary table which contains one row for each span name and data about number
+   * of running and sampled spans.
+   *
+   * <p>Clicking on a cell in the table with a number that is greater than zero will display
+   * detailed information about that span.
+   *
+   * <p>This method will add the TracezSpanProcessor to the tracerProvider, it should only be called
+   * once.
+   *
+   * @return a {@code ZPageHandler} for tracing debug.
+   */
+  public static ZPageHandler getTracezZPageHandler() {
+    addTracezSpanProcessor();
+    return tracezZPageHandler;
+  }
+
+  /**
+   * Registers all zPages to the given {@link HttpServer} {@code server}.
+   *
+   * @param server the server that exports the zPages.
+   */
+  public static void registerAllPagesToHttpServer(HttpServer server) {
```

Comment:
I think this method can be private, yes?

Reply:
I think it should be public. User should be able to call this function to register the zpages to their own server

Reply:
Hm. There are 3 public methods on this class. Do we need all 3? I see 2 of them documented in the javadoc, so maybe the 3rd (getTracezZpageHandler()) can be private?

Reply:
I've updated `getTracezZPageHandler` to `registerTracezZPageHandler` and it can be private, I made it public initially because I wanted to test the "add span processor only once" behavior, should I make it private and get rid of the test?

Reply:
If you want to test that behavior, then make it package-access, and mark it as `@VisibleForTesting`. Thanks!